.github/workflows/test.yml

```diff
@@ -0,0 +1,110 @@
+  NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+  NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+  NUGET_PUBLISH_KEY: ${{ secrets.NUGET_PUBLISH_KEY }}
+  PROVIDER: pulumiservice
+  PUBLISH_REPO_PASSWORD: ${{ secrets.OSSRH_PASSWORD }}
+  PUBLISH_REPO_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+  PULUMI_ACCESS_TOKEN: ${{ secrets.PULUMI_ACCESS_TOKEN }}
+  PULUMI_BACKEND_URL: https://api.pulumi-staging.io
+  PULUMI_GO_DEP_ROOT: ${{ github.workspace }}/..
+  PULUMI_JAVA_SDK_VERSION: 0.10.0
+  PULUMI_LOCAL_NUGET: ${{ github.workspace }}/nuget
+  PULUMI_TEST_OWNER: service-provider-test-org
+  PULUMI_TEST_USE_SERVICE: "true"
+  PYPI_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
+  PYPI_USERNAME: __token__
+  SIGNING_KEY: ${{ secrets.JAVA_SIGNING_KEY }}
+  SIGNING_KEY_ID: ${{ secrets.JAVA_SIGNING_KEY_ID }}
+  SIGNING_PASSWORD: ${{ secrets.JAVA_SIGNING_PASSWORD }}
+  TF_APPEND_USER_AGENT: pulumi
+
+jobs:
+  test:
+    if: github.event_name == 'repository_dispatch' ||
+      github.event.pull_request.head.repo.full_name == github.repository
+    name: test
+    permissions:
+      contents: read
+      id-token: write
+    runs-on: ubuntu-latest
+    env:
+      PROVIDER_VERSION: ${{ inputs.version }}
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ env.PR_COMMIT_SHA }}
+        persist-credentials: false
+    - name: Setup tools
+      uses: ./.github/actions/setup-tools
+      with:
+        tools: pulumictl, pulumicli, go, nodejs, python, dotnet, java
+    - name: Download bin
+      uses: ./.github/actions/download-bin
+    - name: Add NuGet source
+      run: dotnet nuget add source ${{ github.workspace }}/nuget
+    - name: Download nodejs SDK
+      uses: ./.github/actions/download-sdk
+      with:
+        language: nodejs
+    - name: Download python SDK
+      uses: ./.github/actions/download-sdk
+      with:
+        language: python
+    - name: Download dotnet SDK
+      uses: ./.github/actions/download-sdk
+      with:
+        language: dotnet
+    - name: Download go SDK
+      uses: ./.github/actions/download-sdk
+      with:
+        language: go
+    - name: Download java SDK
+      uses: ./.github/actions/download-sdk
+      with:
+        language: java
+    - name: Update path
+      run: echo "${{ github.workspace }}/bin" >> "$GITHUB_PATH"
+    - name: Install Python deps
+      run: |-
+        pip3 install virtualenv==20.0.23
+        pip3 install pipenv
+    - name: Install dependencies
+      run: make install_sdks
+
+    - name: Generate shard
+      run: make shard >> "$GITHUB_ENV"
+      env:
+        TOTAL: ${{ matrix.total }}
+        INDEX: ${{ matrix.index }}
+    - name: Run example tests
+      run: make test_shard
+    - name: Upload coverage reports to Codecov
+      uses: codecov/codecov-action@v4
+      env:
+        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+    strategy:
+      fail-fast: false
+      matrix:
+        total:
+        - 6
+        index:
+        - 0
```

Comment:
This sharding of tests is awesome! Is it possible to NOT run them in parallel within a shard though? As far as I understood, that's why the logs are jumbled up like in example below. I guess it would be losing speed, but debugging any failures would be easier.

```
=== NAME  TestGoTeamsExample
    command.go:121: Command completed without output
    command.go:46: **** Invoke '/opt/hostedtoolcache/pulumi/3.142.1-alpha.xbe2d2a5/x64/pulumi stack import --file stack.json' in '/home/runner/go/src/stackName-1733429656837535781-118258'
=== NAME  TestYamlSchedulesExample
    command.go:118: Wrote output to /tmp/test-stack-22792-510911093/command-output/pulumi-stack-rm.20241205-201450.c9ae5.log
```

Reply:
We're actually in the process of removing `gotestfmt` because it doesn't handle parallel output correctly https://github.com/pulumi/ci-mgmt/pull/1069

Instead of worrying about this CI output we're trying to move in a direction where it's significantly easier to reproduce CI failures locally.